### DISCUSSION TITLE
How to inject a raw property

### DISCUSSION THREAD

**Discussion by jaxoncreed** [LinkedSoftwareDependencies/Components.js#40] [Q&A] [answered]
I'd like to inject a raw property into my component (a string, number, or raw JSON variable). For example, I have this component:

```typescript
interface SomeJsonInterface {
  a: string;
  b: {
    c: string;
    d: number;
  }
  e: number[];
}

interface SomeComponentArgs {
  someString: string;
  someNumber: number;
  someJson: SomeJsonInterface;
}

export class SomeComponent {
  constructor(args: SomeComponentArgs) {
    // ...
  }
}
```

How would I wire up this component in the configuration?

**Comment by rubensworks** [comment, marked as answer]
I assume you're using the generator in the solid server.
If so, it will generate parameters for all fields (`SomeComponent:_someString`, `SomeComponent:_someNumber`, `SomeComponent:_someJson_a`, ...)
You can find the exact param identifiers inside the autogenerated `components` folder.

In that case, instantiating it via a config file happens in [the standard way](https://github.com/solid/community-server/blob/master/config/presets/init.json#L16-L20).
So here this would look something like:
```json
{
  "@id": "urn:something",
  "@type": "SomeComponent",
  "SomeComponent:_someString": "abc",
  "SomeComponent:_someNumber": 123,
  "SomeComponent:_someJson_a": "def",
  ...
}
```

**Comment by jaxoncreed** [comment]
Is there no way to provide so nice raw JSON? Do I have to do it via underscores?

**Comment by rubensworks** [comment]
No, there's no way to do this at the moment.

But should be perfectly possible to implement this though.
This would require some small changes in both Components.js, and the generator (setting `@range JSON` on `someJson` in jsdoc).

**Reply by rubensworks** [reply]
For future reference, this is possible now via `@range {json}`
